packages/sources/tp-icap/src/endpoint/price.ts

```diff
@@ -0,0 +1,128 @@
+        logger.debug('Opening WS connection')
+
+        return new Promise((resolve, reject) => {
+          connection.on('message', (data: WebSocketRawData) => {
+            const { msg, sta } = JSON.parse(data.toString())
+            if (msg === 'auth' && sta === 1) {
+              logger.info('Got logged in response, connection is ready')
+              providerDataStreamEstablishedUnixMs = Date.now()
+              resolve()
+            } else {
+              reject(new Error('Failed to make WS connection'))
+            }
+          })
+          const options = {
+            msg: 'auth',
+            user: WS_API_USERNAME,
+            pass: WS_API_PASSWORD,
+            mode: 'broadcast',
+          }
+          connection.send(JSON.stringify(options))
+        })
+      },
+      message: (message) => {
+        logger.debug({ msg: 'Received message from WS', message })
+
+        const providerDataReceivedUnixMs = Date.now()
+
+        if (!('msg' in message) || message.msg === 'auth') return []
+
+        const { fvs, rec, sta } = message
+
+        if (!fvs || !rec || sta !== 1) {
+          logger.error({ msg: 'Missing expected field `fvs` or `rec` from `sub` message', message })
+          return []
+        }
+
+        const { ACTIV_DATE, ASK, BID, MID_PRICE, TIMACT } = fvs
+
+        const base = rec.slice(5, 8)
+        const quote = rec.slice(8, 11)
+
+        if (!isNum(MID_PRICE) && !(isNum(BID) && isNum(ASK))) {
+          const errorMessage = 'TP ICAP `sub` message did not include required price fields'
+          logger.debug({ errorMessage })
+          return []
+        }
+
+        const result =
+          MID_PRICE ||
+          new Decimal(ASK as number)
+            .add(BID as number)
+            .div(2)
+            .toNumber()
+
+        const response = {
+          params: {
+            base,
+            quote,
+          },
+          response: {
+            result,
+            data: {
+              result,
+            },
+            timestamps: {
+              providerDataReceivedUnixMs,
+              providerDataStreamEstablishedUnixMs,
+              providerIndicatedTimeUnixMs:
+                ACTIV_DATE && TIMACT ? new Date(ACTIV_DATE + ' ' + TIMACT).getTime() : undefined,
+            },
+          },
+        }
+
+        const responseWithRec = {
+          params: {
+            base: rec, // Cache the price with the full rec symbol as `base` as a separate entry
+            quote,
+          },
+          response: {
+            result,
+            data: {
+              result,
+            },
+            timestamps: {
+              providerDataReceivedUnixMs,
+              providerDataStreamEstablishedUnixMs,
+              providerIndicatedTimeUnixMs:
+                ACTIV_DATE && TIMACT ? new Date(ACTIV_DATE + ' ' + TIMACT).getTime() : undefined,
+            },
+          },
+        }
```

Comment:
While this certainly works, my recommendation would be to add a `requestTransform` (or overrides) that will always convert the parameters into a consistent one, either separate or the full provider value. This way we have only one entry, and it makes it easier to debug in the future